backend/src/user/user.controller.ts

```diff
@@ -50,4 +59,20 @@ export class UserController {
     }
     await this.userService.updateUserDetails(userId, data);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Get(":userId(\\d+)")
+  async getUserName(
+    @Request() req: ExpressRequest,
+    @Response() res: ExpressResponse,
+    @Param("userId", ParseIntPipe) userId: number,
+  ): Promise<GetUserNameRes | null> {
+    const User = await this.userService.getNameById(userId);
+    const responseBody: GetUserNameRes = {
+      name: User?.name ?? "",
+    };
+
+    res.json(responseBody);
+    return responseBody;
```

Comment:
We can just return the `GetUserNameRes` or `null` object and we don't have to call `res.json` ourselves (as long as we don't inject the `@Response` object). You can read more about it in this section https://docs.nestjs.com/controllers#request-object.